spec/controllers/users_controller_spec.rb

```diff
@@ -240,7 +240,7 @@
     end
 
     context 'timezone_filter is blank' do
```

Comment:
@tansaku do you have any suggestions for changing the way this is being tested?

Reply:
@nisevi looks to me like these tests are unrelated to the paper_trail gem  - I think we also don't really use the paper_trail gem - so it could be removed.

Refactoring these tests should be the subject of another PR I think, but we should also make sure that we are not refactoring for refactoring's sake.  The critical thing is that we deliver value to users with features, and which features we want to deliver affect the best choices for refactoring.

Easiest path is probably to just upgrade papertrail and then open a separate ticket about refactoring the tests ...

Reply:
@tansaku perfect, I will remove `paper_trail` with this PR then :+1: .

I commented out those tests because after the upgrade they were all failing. So I think that they are related but I don't know in which way.

Next step that I will do is to remove this gem locally, if all tests are passing then we are sure that we can remove it without any side effect.

Reply:
@tansaku when trying to remove `paper_trail` I found `app/models/concerns/act_as_page.rb`:

```
module ActAsPage
  def self.included(base)
    base.acts_as_tree
    base.extend FriendlyId
    base.friendly_id :title, use: :slugged
    base.has_paper_trail
    base.validates :title, presence: true
  end
end
```

Reply:
ah, my mistake - sorry confusing it with another plugin

Reply:
but the "timezone_filter is blank" test for the user controller is completely unrelated to ActAsPage/paper_trail right?

Reply:
@tansaku it is not trivial to remove this gem.

Tests failing after removing:
```
base.has_paper_trail
```

on `app/models/concerns/act_as_page.rb`

```
rspec ./spec/models/document_spec.rb:6 # Document 
rspec ./spec/models/document_spec.rb:17 # Document is invalid without title
rspec ./spec/models/document_spec.rb:13 # Document is valid with proper attributes
rspec ./spec/models/document_spec.rb:7 # Document 
rspec ./spec/models/document_spec.rb:9 # Document has public-activity enabled
rspec ./spec/models/document_spec.rb:21 # Document is invalid without project
rspec ./spec/models/document_spec.rb:26 # Document#url_for_me returns correct url for show action
rspec ./spec/models/document_spec.rb:30 # Document#url_for_me returns correct url for other actions
rspec ./spec/models/document_spec.rb:36 # Document#slug_candidates returns correct slug candidates
rspec './spec/requests/authentications_spec.rb[1:1:1:1]' # OmniAuth authentication for unregistered users with a GitHub profile should work with valid credentials
rspec './spec/requests/authentications_spec.rb[1:1:1:3]' # OmniAuth authentication for unregistered users with a GitHub profile should not allow removal of profiles without passwords
rspec './spec/requests/authentications_spec.rb[1:3:1:3]' # OmniAuth authentication for unregistered users with a Google profile should not allow removal of profiles without passwords
rspec './spec/requests/authentications_spec.rb[1:3:1:1]' # OmniAuth authentication for unregistered users with a Google profile should work with valid credentials
rspec './spec/requests/authentications_spec.rb[1:2:1:3]' # OmniAuth authentication for registered users with a GitHub profile should be able to create other profiles
rspec './spec/requests/authentications_spec.rb[1:2:1:1]' # OmniAuth authentication for registered users with a GitHub profile finds the right user if auth exists
rspec './spec/requests/authentications_spec.rb[1:2:1:4]' # OmniAuth authentication for registered users with a GitHub profile should not accept multiple profiles from the same source
rspec './spec/requests/authentications_spec.rb[1:2:1:2]' # OmniAuth authentication for registered users with a GitHub profile should be removable for users with a password
rspec './spec/requests/authentications_spec.rb[1:4:1:1]' # OmniAuth authentication for registered users with a Google profile finds the right user if auth exists
rspec './spec/requests/authentications_spec.rb[1:4:1:2]' # OmniAuth authentication for registered users with a Google profile should be removable for users with a password
rspec './spec/requests/authentications_spec.rb[1:4:1:3]' # OmniAuth authentication for registered users with a Google profile should be able to create other profiles
rspec './spec/requests/authentications_spec.rb[1:4:1:4]' # OmniAuth authentication for registered users with a Google profile should not accept multiple profiles from the same source
rspec ./spec/models/static_page_spec.rb:6 # StaticPage should be versioned
rspec ./spec/controllers/concerns/statistics_spec.rb:35 # StatisticsConcern gets stats for documents
rspec ./spec/controllers/documents_controller_spec.rb:28 # DocumentsController should raise an error if no project was found
rspec ./spec/controllers/documents_controller_spec.rb:129 # DocumentsController POST create with valid params creates a new Document
rspec ./spec/controllers/documents_controller_spec.rb:141 # DocumentsController POST create with valid params redirects to the created document
rspec ./spec/controllers/documents_controller_spec.rb:135 # DocumentsController POST create with valid params assigns a newly created document as @document
rspec ./spec/controllers/documents_controller_spec.rb:146 # DocumentsController POST create with valid params creates a document create activity
rspec ./spec/controllers/documents_controller_spec.rb:154 # DocumentsController POST create with invalid params assigns a newly created but unsaved document as @document
rspec ./spec/controllers/documents_controller_spec.rb:162 # DocumentsController POST create with invalid params re-renders the new template
rspec ./spec/controllers/documents_controller_spec.rb:121 # DocumentsController GET new renders the new template
rspec ./spec/controllers/documents_controller_spec.rb:117 # DocumentsController GET new assigns a new document as @document
rspec ./spec/controllers/documents_controller_spec.rb:86 # DocumentsController PUT update_document_parent_id/ with a valid parent id changes the document parent id
rspec ./spec/controllers/documents_controller_spec.rb:91 # DocumentsController PUT update_document_parent_id/ with a valid parent id assigns flash message after changing parent_id
rspec ./spec/controllers/documents_controller_spec.rb:107 # DocumentsController PUT update_document_parent_id/ with an invalid parent id renders a flash error message
rspec ./spec/controllers/documents_controller_spec.rb:100 # DocumentsController PUT update_document_parent_id/ with an invalid parent id does not change the document parent id
rspec ./spec/controllers/documents_controller_spec.rb:65 # DocumentsController get_doc_categories it has categories to show renders the categories partial
rspec ./spec/controllers/documents_controller_spec.rb:70 # DocumentsController get_doc_categories it has categories to show assigns the available categories to @categories
rspec ./spec/controllers/documents_controller_spec.rb:55 # DocumentsController GET show with more than one project should not mistakenly render the document under the wrong project
rspec ./spec/controllers/documents_controller_spec.rb:45 # DocumentsController GET show with a single project renders the show template
rspec ./spec/controllers/documents_controller_spec.rb:41 # DocumentsController GET show with a single project assigns the requested document as @document
rspec ./spec/controllers/documents_controller_spec.rb:222 # DocumentsController POST mercury_update with valid params should create a document update activity
rspec ./spec/controllers/documents_controller_spec.rb:211 # DocumentsController POST mercury_update with valid params should render an empty string
rspec ./spec/controllers/documents_controller_spec.rb:216 # DocumentsController POST mercury_update with valid params should update the document with the new title and body
rspec ./spec/controllers/documents_controller_spec.rb:174 # DocumentsController DELETE destroy destroys the requested document
rspec ./spec/controllers/documents_controller_spec.rb:180 # DocumentsController DELETE destroy redirects to the documents list
rspec ./spec/requests/events_spec.rb:14 # Events allows prepopulation of form with category
rspec ./spec/requests/events_spec.rb:9 # Events allows prepopulation of form with name
rspec ./spec/requests/events_spec.rb:19 # Events allows prepopulation of form with project
rspec ./spec/helpers/documents_helper_spec.rb:10 # DocumentsHelper#metadata should return metadata of the newly created document
rspec ./spec/helpers/documents_helper_spec.rb:15 # DocumentsHelper#metadata should return metadata of the latest version of the document
```